### PR TITLE
feat: update to use new ConversionService

### PR DESCRIPTION
**Pull request by sdelamo**
This still fails with 

```

io.micronaut.rxjava3.http.client.RxClientCloseSpec confirm RxHttpClient can be stopped FAILED (1s)

  java.lang.IllegalStateException: No JsonMapper implementation found
      at io.micronaut.json.JsonMapper.lambda$createDefault$1(JsonMapper.java:246)
      at java.base/java.util.Optional.orElseThrow(Optional.java:403)
      at io.micronaut.json.JsonMapper.createDefault(JsonMapper.java:246)
      at io.micronaut.http.client.netty.DefaultHttpClient.createDefaultMediaTypeRegistry(DefaultHttpClient.java:1844)
      at io.micronaut.http.client.netty.DefaultHttpClient.<init>(DefaultHttpClient.java:409)
      at io.micronaut.http.client.netty.DefaultHttpClient.<init>(DefaultHttpClient.java:391)
      at io.micronaut.http.client.netty.NettyHttpClientFactory.createNettyClient(NettyHttpClientFactory.java:128)
      at io.micronaut.http.client.netty.NettyHttpClientFactory.createNettyClient(NettyHttpClientFactory.java:113)
      at io.micronaut.http.client.netty.NettyHttpClientFactory.createClient(NettyHttpClientFactory.java:54)
      at io.micronaut.http.client.HttpClient.create(HttpClient.java:245)
      at io.micronaut.rxjava3.http.client.Rx3HttpClient.create(Rx3HttpClient.java:115)
      at io.micronaut.rxjava3.http.client.RxClientCloseSpec.confirm RxHttpClient can be stopped(RxClientCloseSpec.groovy:9)

io.micronaut.rxjava3.http.client.RxClientCloseSpec confirm RxHttpClient can be closed FAILED

  java.lang.IllegalStateException: No JsonMapper implementation found
      at io.micronaut.json.JsonMapper.lambda$createDefault$1(JsonMapper.java:246)
      at java.base/java.util.Optional.orElseThrow(Optional.java:403)
      at io.micronaut.json.JsonMapper.createDefault(JsonMapper.java:246)
      at io.micronaut.http.client.netty.DefaultHttpClient.createDefaultMediaTypeRegistry(DefaultHttpClient.java:1844)
      at io.micronaut.http.client.netty.DefaultHttpClient.<init>(DefaultHttpClient.java:409)
      at io.micronaut.http.client.netty.DefaultHttpClient.<init>(DefaultHttpClient.java:391)
      at io.micronaut.http.client.netty.NettyHttpClientFactory.createNettyClient(NettyHttpClientFactory.java:128)
      at io.micronaut.http.client.netty.NettyHttpClientFactory.createNettyClient(NettyHttpClientFactory.java:113)
      at io.micronaut.http.client.netty.NettyHttpClientFactory.createClient(NettyHttpClientFactory.java:54)
      at io.micronaut.http.client.HttpClient.create(HttpClient.java:245)
      at io.micronaut.rxjava3.http.client.Rx3HttpClient.create(Rx3HttpClient.java:115)
      at io.micronaut.rxjava3.http.client.RxClientCloseSpec.confirm RxHttpClient can be closed(RxClientCloseSpec.groovy:24)

io.micronaut.rxjava3.http.client.RxJavaClientInjectSpec initializationError FAILED

  io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [jacksonConfiguration] of class: io.micronaut.http.server.exceptions.response.HateoasErrorResponseProcessor
  
  Message: No bean of type [io.micronaut.json.JsonConfiguration] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the 'micronaut-inject-java' dependency should be configured as an annotation processor).
  Path Taken: NettyEmbeddedServer.buildDefaultServer(NettyHttpServerConfiguration configuration) --> new DefaultNettyEmbeddedServerFactory(ApplicationContext applicationContext,[RouteExecutor routeExecutor],MediaTypeCodecRegistry mediaTypeCodecRegistry,StaticResourceResolver staticResourceResolver,ThreadFactory nettyThreadFactory,HttpCompressionStrategy httpCompressionStrategy,EventLoopGroupFactory eventLoopGroupFactory,EventLoopGroupRegistry eventLoopGroupRegistry,WebSocketUpgradeHandlerFactory webSocketUpgradeHandlerFactory) --> new RouteExecutor(Router router,BeanContext beanContext,RequestArgumentSatisfier requestArgumentSatisfier,HttpServerConfiguration serverConfiguration,[ErrorResponseProcessor errorResponseProcessor],ExecutorSelector executorSelector) --> new HateoasErrorResponseProcessor([JsonConfiguration jacksonConfiguration])
      at app//io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:2086)
      at app//io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConst
```

### CODE REVIEW

**Comment by sdelamo** [comment]
@wetted can you work on this issue? see https://github.com/micronaut-projects/micronaut-reactor/pull/237

**Comment by wetted** [comment]
@sdelamo this one should be good to go now.

File: rxjava3/src/main/resources/META-INF/services/io.micronaut.core.convert.TypeConverterRegistrar [hunk 1]
```diff
@@ -0,0 +1 @@
+io.micronaut.rxjava3.converters.RxJava3ConverterRegistrar
```
Comment (sdelamo) [inline rxjava3/src/main/resources/META-INF/services/io.micronaut.core.convert.TypeConverterRegistrar:1]: if we keep the `@Singleton` annotation in `RxJava3ConverterRegistrar` we don't need this.

Comment (sdelamo) [reply rxjava3/src/main/resources/META-INF/services/io.micronaut.core.convert.TypeConverterRegistrar:1]: I have removed it https://github.com/micronaut-projects/micronaut-rxjava3/pull/226/commits/2c002c94c3d595e667300e87cd032839526c8486

**Pull request merged**